src/api/app/controllers/webui/users/tokens_controller.rb

```diff
@@ -32,9 +32,7 @@ def update
         end
       end
       format.html do
-        new_scm_token = params.require(:token).except(:string_readonly).permit(:scm_token)
-
-        if @token.update(new_scm_token)
+        if @token.update(update_parameters)
           flash[:success] = 'Token successfully updated'
         else
           flash[:error] = 'Failed to update Token'
```

Comment:
If something goes wrong, users won't know what went wrong. We should include error messages in the `flash`.

```suggestion
          flash[:error] = "Failed to update Token: #{@token.errors.full_messages.to_sentence}"
```

Reply:
True. I also changed the error message in the case of regenerating the SCM string.